docs/resources/vault_secrets_dynamic_secret.md

```diff
@@ -0,0 +1,73 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "hcp_vault_secrets_dynamic_secret Resource - terraform-provider-hcp"
+subcategory: ""
+description: |-
+  The Vault Secrets dynamic secret resource manages a dynamic secret configuration.
+---
+
+# hcp_vault_secrets_dynamic_secret (Resource)
+
+The Vault Secrets dynamic secret resource manages a dynamic secret configuration.
+
+## Example Usage
+
+```terraform
+resource "hcp_vault_secrets_dynamic_secret" "example_aws" {
+  app_name         = "my-app-1"
+  secret_provider  = "aws"
```

Comment:
Provider is a reserved and very meaningful term in TF-land so I went with `secret_provider` to disambiguate. This is the term that was used for the [data sources](https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/data-sources/vault_secrets_rotating_secret).